pkg/services/alerting/extractor.go

```diff
@@ -193,8 +194,30 @@ func (e *DashAlertExtractor) getAlertFromPanels(jsonWithPanels *simplejson.Json,
 			return nil, err
 		}
 
-		if !validateAlertFunc(alert) {
-			return nil, ValidationError{Reason: fmt.Sprintf("Panel id is not correct, alertName=%v, panelId=%v", alert.Name, alert.PanelId)}
+		validationErrors := strings.Builder{}
+		validationWarnings := strings.Builder{}
+		for _, validator := range validators {
+			ok, reason := validator.aFunc(alert)
+			if !ok {
+				switch validator.aSeverity {
+				case alertError:
+					if validationErrors.Len() > 0 {
+						validationErrors.WriteString("\n")
+					}
+					validationErrors.WriteString(reason)
+				case alertWarning:
+					if validationWarnings.Len() > 0 {
+						validationErrors.WriteString("\n")
+					}
+					validationWarnings.WriteString(reason)
+				}
+			}
```

Comment:
It's generally more readable, and idiomatic, if you reduce nesting:
```suggestion
			if ok {
				continue
			}
			
			switch validator.aSeverity {
			case alertError:
				if validationErrors.Len() > 0 {
					validationErrors.WriteString("\n")
				}
				validationErrors.WriteString(reason)
			case alertWarning:
				if validationWarnings.Len() > 0 {
					validationErrors.WriteString("\n")
				}
				validationWarnings.WriteString(reason)
			}
```